ledger_test.go

```diff
@@ -124,3 +124,39 @@ func TestLedger_AddTransaction(t *testing.T) {
 		t.Fatal("more than 1 round finalized")
 	}
 }
+
+func TestLedger_CallExpensiveContract(t *testing.T) {
+	testnet := NewTestNetwork(t)
+	defer testnet.Cleanup()
+
+	testnet.AddNode(t, 500000)
+	bob := testnet.AddNode(t, 100000000)
+
+	testnet.WaitForLatestConsensus(t)
+
+	registry, err := testnet.faucet.SpawnContract("testdata/registry.wasm",
+		100000000, nil)
+	assert.NoError(t, err)
+
+	testnet.WaitForLatestConsensus(t)
+
+	contract, err := testnet.faucet.SpawnContract("testdata/registered_contract.wasm",
+		100000000, nil)
+	assert.NoError(t, err)
+
+	testnet.WaitForLatestConsensus(t)
+
+	// Deposit some gas to both smart contracts
+	_, err = testnet.faucet.DepositGas(registry.ID, 100000000)
+	assert.NoError(t, err)
+	_, err = testnet.faucet.DepositGas(contract.ID, 100000000)
+	assert.NoError(t, err)
+	testnet.WaitForLatestConsensus(t)
+
+	_, err = bob.CallContract(registry.ID, 0, 1, "track", contract.ID[:])
```

Comment:
The issue I'm having right now is that I couldn't setup the test case to replicate the issue. According to @ryanswart, calling the track method with 0 amount and 1 gas limit, while the caller has 1mil PERL should succeed, but it always fails in the test case for some reason. And it ends up using more than 2 PERL as transaction fee.